### DISCUSSION TITLE
Can rigid bodies move beyond the degrees of freedom of joints？

### DISCUSSION THREAD

**Discussion by GhostMinerPlus** [dimforge/rapier#722] [Q&A] [answered]
While I use FixedJoint and RevoluteJoint to bind two rigid bodies, they still can move from each other if I add a big force on one of them.

**Comment by ThierryBerger** [comment, marked as answer]
When using `ImpulseJointSet`, the approach to resolving constraints is iterative, which means there can be "errors", or approximations.

This behaviour can be improved by tweaking rapier configuration (increasing for example `IntegrationParameters::num_solver_iterations` or lowering `IntegrationParameters::dt`).

But if you're looking for a robust simulation of joints, you'll want to use `MultibodyJointSet`.

I'm converting to discussion, so you can mark it as answered if satisfying.

**Reply by GhostMinerPlus** [reply]
I tried `MultibodyJoint` in bevy, but it did not work.

**Reply by GhostMinerPlus** [reply]
It seems that I misunderstood `MultibodyJoint`'s usage before, but I didn't find any useful docs about `MultibodyJoint`.
